mir_eval/segment.py

```diff
@@ -165,9 +177,20 @@ def validate_structure(
         if not np.allclose(reference_intervals.max(), estimated_intervals.max()):
             raise ValueError("End times do not match")
 
+def _default_on_empty(default_return_value):
+    def decorator_default_on_empty(func):
+        @functools.wraps(func)
+        def wrapper_default_on_empty(reference_indices, estimated_indices, *args, **kwargs):
+            # Check for empty annotations.  Don't need to check labels because
+            # validate_structure makes sure they're the same size as intervals
+            if len(reference_indices) == 0 or len(estimated_indices) == 0:
+                return default_return_value
+            return func(reference_indices, estimated_indices, *args, **kwargs)
+        return wrapper_default_on_empty
+    return decorator_default_on_empty
```

Comment:
This would probably be cleaner using the `decorator` module, as we do in util, eg https://github.com/mir-evaluation/mir_eval/blob/935451d7c06b6eea63608d4b8a869f333440b1df/mir_eval/util.py#L946-L962

Reply:
Good note, gotta update my solution!